src/hotspot/share/services/memoryManager.cpp

```diff
@@ -293,7 +293,8 @@ void GCMemoryManager::gc_end(bool recordPostGCUsage,
     }
 
     if (is_notification_enabled()) {
-      GCNotifier::pushNotification(this, _gc_end_message, GCCause::to_string(cause));
+      const char* message = notificationMessage != nullptr ? notificationMessage : _gc_end_message;
+      GCNotifier::pushNotification(this, message, GCCause::to_string(cause));
```

Comment:
This looks great to me. I am assuming all collectors that do their work in multiple pieces (at least ZGC and possibly G1 too, may be?) might want to make use of this functionality.

I am guessing this might benefit from an upstream review at some point.

Reply:
Yes, I reckon so. I will target these changes for upstream as well.